tests/functional/codegen/features/test_transient.py

```diff
@@ -55,3 +44,316 @@ def setter(k: address, v: uint256):
 
     assert "TLOAD" in t
     assert "TSTORE" in t
+
+
+@pytest.mark.transient
+@pytest.mark.parametrize(
+    "typ,value,zero",
+    [
+        ("uint256", 42, 0),
+        ("int256", -(2**200), 0),
+        ("int128", -(2**126), 0),
+        ("address", "0xEeeeeEeeeEeEeeEeEeEeeEEEeeeeEeeeeeeeEEeE", None),
+        ("bytes32", b"deadbeef" * 4, b"\x00" * 32),
+        ("bool", True, False),
+        ("String[10]", "Vyper hiss", ""),
+        ("Bytes[10]", b"Vyper hiss", b""),
+    ],
+)
+def test_value_storage_retrieval(typ, value, zero, get_contract):
+    code = f"""
+bar: public(transient({typ}))
+
+@external
+def foo(a: {typ}) -> {typ}:
+    self.bar = a
+    return self.bar
+    """
+
+    c = get_contract(code)
+    assert c.bar() == zero
+    assert c.foo(value) == value
+    assert c.bar() == zero
+    assert c.foo(value) == value
+    assert c.bar() == zero
+
+
+@pytest.mark.transient
+@pytest.mark.parametrize("val", [0, 1, 2**256 - 1])
+def test_usage_in_constructor(get_contract, val):
+    code = """
+A: transient(uint256)
+a: public(uint256)
+
+
+@deploy
+def __init__(_a: uint256):
+    self.A = _a
+    self.a = self.A
+
+
+@external
+@view
+def a1() -> uint256:
+    return self.A
+    """
+
+    c = get_contract(code, val)
+    assert c.a() == val
+    assert c.a1() == 0
+
+
+def test_multiple_transient_values(get_contract):
+    code = """
+a: public(transient(uint256))
+b: public(transient(address))
+c: public(transient(String[64]))
+
+@external
+def foo(_a: uint256, _b: address, _c: String[64]) -> (uint256, address, String[64]):
+    self.a = _a
+    self.b = _b
+    self.c = _c
+    return self.a, self.b, self.c
+    """
+    values = (3, "0xEeeeeEeeeEeEeeEeEeEeeEEEeeeeEeeeeeeeEEeE", "Hello world")
+
+    if version_check(begin="cancun"):
```

Comment:
use the marker

Reply:
maybe
```
with pytest.raises(ExceptionList) as exceptions:
    compile_code(...)
# ExceptionList subclasses list
assert len(set(type(e) for e in exceptions.value))) == 1
raise exceptions.value[0]

Reply:
maybe something like
```python
try:
    compile_code(...)
except ExceptionList as exceptions:
    # ExceptionList subclasses list
    # check they are all same type
    assert len(set(type(e) for e in exceptions.value))) == 1
    # 
    raise exceptions.value[0]
```

Reply:
This would not work because `ExceptionList` is not of `Exception` type and so it cannot be caught. The workaround I can currently think of is to assert that `TransientStorageException` string appears thrice in the error message of the generic `VyperException`.